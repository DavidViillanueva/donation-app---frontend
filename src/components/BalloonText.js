import React from 'react';

import './BallonText.css';

const BalloonText = ({ amount = 0,actualAmount = 0 }) => {
    return (
        <div className="text-alert">
            <div className="indicator">
                <span> ${ amount-actualAmount } still needed for this project!</span>
            </div>
            <div className="pointer"></div>
        </div>
    )
};

export default BalloonText;
