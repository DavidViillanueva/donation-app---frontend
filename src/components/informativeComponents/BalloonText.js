import React,{ useEffect }  from 'react';
import { useTranslation } from 'react-i18next'


import './BallonText.css';

const BalloonText = ({ amount = 0,actualAmount = 0 ,lang }) => {

    const { t, i18n } = useTranslation();

    useEffect(() => {
        i18n.changeLanguage(lang);
    }, [lang,i18n])

    return (
        <div className="text-alert">
            <div className="indicator">
            <span> {t('balloon',{restantAmount: amount-actualAmount })}</span>
            </div>
            <div className="pointer"></div>
        </div>
    )
};

export default BalloonText;
