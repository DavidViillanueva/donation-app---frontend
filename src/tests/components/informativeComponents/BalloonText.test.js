
import React from 'react';
import { shallow } from 'enzyme';
import BalloonText from '../../../components/informativeComponents/BalloonText';

describe('BallonTex component', () => {
    test('should render correctly', () => {
        const wrapper = shallow(<BalloonText />);

        expect( wrapper ).toMatchSnapshot();
    })
    
})
