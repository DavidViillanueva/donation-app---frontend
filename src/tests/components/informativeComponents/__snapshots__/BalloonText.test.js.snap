// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BallonTex component should render correctly 1`] = `
<div
  className="text-alert"
>
  <div
    className="indicator"
  >
    <span>
       
      balloon
    </span>
  </div>
  <div
    className="pointer"
  />
</div>
`;
